src/SFA.DAS.AssessorService.Application/Handlers/Private/StartPrivateCertificateHandler.cs

```diff
@@ -0,0 +1,105 @@
+using SFA.DAS.AssessorService.Domain.JsonData;
+using SFA.DAS.AssessorService.ExternalApis.AssessmentOrgs;
+
+namespace SFA.DAS.AssessorService.Application.Handlers.Private
+{
+    public class StartPrivateCertificateHandler : IRequestHandler<StartCertificatePrivateRequest, Certificate>
+    {
+        private readonly ICertificateRepository _certificateRepository;
+        private readonly IIlrRepository _ilrRepository;
+        private readonly IAssessmentOrgsApiClient _assessmentOrgsApiClient;
+        private readonly IOrganisationQueryRepository _organisationQueryRepository;
+        private readonly ILogger<StartCertificateHandler> _logger;
+
+        public StartPrivateCertificateHandler(ICertificateRepository certificateRepository, IIlrRepository ilrRepository, IAssessmentOrgsApiClient assessmentOrgsApiClient,
+            IOrganisationQueryRepository organisationQueryRepository, ILogger<StartCertificateHandler> logger)
+        {
+            _certificateRepository = certificateRepository;
+            _ilrRepository = ilrRepository;
+            _assessmentOrgsApiClient = assessmentOrgsApiClient;
+            _organisationQueryRepository = organisationQueryRepository;
+            _logger = logger;
+        }
+
+        public async Task<Certificate> Handle(StartCertificatePrivateRequest request, CancellationToken cancellationToken)
+        {
+            return await _certificateRepository.GetPrivateCertificate(request.Uln, request.LastName) ??
+                   await CreateNewCertificate(request);
+        }
+
+        private async Task<Certificate> CreateNewCertificate(StartCertificatePrivateRequest request)
+        {
+            //_logger.LogInformation("CreateNewCertificate Before Get Ilr from db");
+            //var ilr = await _ilrRepository.Get(request.Uln, request.StandardCode);
+            //_logger.LogInformation("CreateNewCertificate Before Get Organisation from db");
+            var organisation = await _organisationQueryRepository.GetByUkPrn(request.UkPrn);
+            _logger.LogInformation("CreateNewCertificate Before Get Standard from API");
+            //var standard = await _assessmentOrgsApiClient.GetStandard(ilr.StdCode);
+            //_logger.LogInformation("CreateNewCertificate Before Get Provider from API");
+            ////Provider provider;
+            //trym
+            //{
+            //    provider = await _assessmentOrgsApiClient.GetProvider(ilr.UkPrn);
+            //}
+            //catch (Exception)
+            //{
+            //    // see whether there are any previous certificates with this ukrpn and a ProviderName....
+            //    var previousProviderName = await _certificateRepository.GetPreviousProviderName(ilr.UkPrn);
+            //    provider = previousProviderName != null
+            //        ? new Provider { ProviderName = previousProviderName }
+            //        : new Provider { ProviderName = "Unknown" };
+            //}
```

Comment:
These commented lines can be deleted now. 😁 
And that `"CreateNewCertificate Before Get Standard from API"` log entry.

Reply:
yep:-  I think we are looking at an outdated copy - the comments where removed on friday and I can also see in my local copy they are removed.

Reply:
Hopefully should be ok now.

Reply:
amend bits done